### PR TITLE
Setup CZI Playwright framework

### PR DESCRIPTION
**Pull request by charles-testco**

### CODE REVIEW

File: frontend/eslint/e2e.js [hunk 1]
```diff
@@ -1,3 +1,25 @@
...
+        devDependencies: true,
+        optionalDependencies: true,
+        peerDependencies: true,
+      },
+    ],
+    'no-await-in-loop': 'off',
+    'playwright/no-element-handle': 'off',
+    'playwright/no-wait-for-timeout': 'off',
+    'no-explicit-any': 'off',
+    'import/no-default-export': 'off',
```
[the remark this answers is not in the document]
Comment (charles-testco) [reply frontend/eslint/e2e.js:16]: Its been while and don't remember. Will remove and address in the specific files

File: frontend/eslint/e2e.js [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+    '@typescript-eslint/require-await': 'off',
+    '@typescript-eslint/no-explicit-any': 'off',
+    '@typescript-eslint/no-unsafe-member-access': 'off',
+    '@typescript-eslint/no-unsafe-return': 'off',
```
[the remark this answers is not in the document]
Comment (charles-testco) [reply frontend/eslint/e2e.js]: should be fixed now

**Comment by richaagarwal** [comment]
@charles-testco  Is this PR still actively being worked on, or are yours and Victor's other branches replacing this?  If this is still the main PR to get the framework integrated, just wanted to make sure there was a plan to address the feedback here. Thank you!

File: frontend/e2e/utils/_selectors.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import { SearchSortType } from '@/store/search/constants';
```
[the remark this answers is not in the document]
Comment (charles-testco) [reply frontend/e2e/utils/_selectors.ts:1]: It is temporary, I will eventually merge with the new one we created; just did not want to break tests

File: frontend/e2e/utils/selectors.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  `[placeholder="${placeholder}"]`;
+export const getByDataName = (name: string): string => `[data-name="${name}"]`;
+export const getHasText = (type: string, text: string): string =>
+  `${type}has-text("${text}")`;
+export const getByTypeAndName = (type: string, name: string): string =>
+  `${type}[name="${name}"])`;
+export const getByRole = (role: string): string => `[role="${role}"]`;
+export const getMetadataField = (name: string): string =>
+  `[class="data-table__data column-${name}"]`;
+export const getByLabel = (label: string): string => `[label="${label}"]`;
```
[the remark this answers is not in the document]
Comment (charles-testco) [reply frontend/e2e/utils/selectors.ts:19]: Not in this PR but in others, yes. Removing to keep this PR lean

File: frontend/e2e/tests/home/filter.spec.ts [hunk 5]
```diff
@@ -0,0 +1,124 @@
+import { expect, test } from '@playwright/test';
+import { satisfies } from '@renovate/pep440';
+
+import { testPluginFilter } from '../../utils/filter';
```
[the remark this answers is not in the document]
Comment (charles-testco) [reply frontend/e2e/tests/home/filter.spec.ts:4]: Playwright does not find imports this way and there have been extensive discussions on this. Will need to investigate further. I will suggest we fix this in a separate PR

File: frontend/e2e/setup/local.config.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  ],
+
+  projects: [
+    {
+      name: 'chromium',
+      use: {
+        ...devices['Desktop Chrome'],
+        video: 'on',
+      },
+    },
```
[the remark this answers is not in the document]
Comment (charles-testco) [reply frontend/e2e/setup/local.config.ts:41]: I have updated the configs and tested to work. The heights are just approxemiates but I suppose this shouldn't be relevant

**Comment by charles-testco** [comment]
> @charles-testco Is this PR still actively being worked on, or are yours and Victor's other branches replacing this? If this is still the main PR to get the framework integrated, just wanted to make sure there was a plan to address the feedback here. Thank you!

Hi @richaagarwal  this PR implements the bare framework and should go first. We will have few others that will follow today tomorrow but this is the base and needs to go first. I am looking at the feedbacks and hope to address all major ones by today.

**Comment by charles-testco** [comment]
@codemonkey800 should I merge this PR now?

**Pull request merged**